expression/column.go

```diff
@@ -160,6 +160,10 @@ type Column struct {
 	// Index is used for execution, to tell the column's position in the given row.
 	Index int
 
+	// Flag is a flag comes from model.ColumnInfo. We maintain here mainly because we
+	// want to eliminate ifnull on not null column.
+	Flag uint
```

Comment:
This flag can be maintained in the `Column.RetType.Flag`?

Reply:
What if we have sum(ifnull(a, 0))?  How to handle this case?

Reply:
we can add a `NotNull` flag to its result type.